marklogic/src/main/scala/quasar/physical/marklogic/fs/data.scala

```diff
@@ -109,6 +114,35 @@ object data {
     toXml0(rootStr, rootElem, inner)(ejsonEjson)(data)
   }
 
+  ////
+
+  // xs:base64Binary
+  private val base64 = Prism[String, ImmutableArray[Byte]](
+    s => \/.fromTryCatchNonFatal(Base64.getDecoder.decode(s)).map(ImmutableArray.fromArray).toOption)(
+    (Base64.getEncoder.encodeToString(_)) compose (_.toArray))
+
+  // xs:date
+  private val localDate: Prism[String, LocalDate] = temporal(LocalDate.from, DateTimeFormatter.ISO_LOCAL_DATE)
+  // xs:time
+  private val localTime: Prism[String, LocalTime] = temporal(LocalTime.from, DateTimeFormatter.ISO_LOCAL_TIME)
+  // xs:dateTime
+  private val instant: Prism[String, Instant] = temporal(Instant.from, DateTimeFormatter.ISO_INSTANT)
+
+  private def temporal[T <: TemporalAccessor](f: TemporalAccessor => T, fmt: DateTimeFormatter): Prism[String, T] =
+    Prism[String, T](s => \/.fromTryCatchNonFatal(f(fmt.parse(s))).toOption)(fmt.format)
+
+  private val DurationEncoding = "(-?\\d+)(?:\\.(\\d+))?".r
+
+  private val duration = Prism[String, Duration] {
+    case DurationEncoding(secs, nanos) =>
+      \/.fromTryCatchNonFatal(Duration.ofSeconds(secs.toLong, nanos.toLong)).toOption
+
+    case DurationEncoding(secs) =>
+      \/.fromTryCatchNonFatal(Duration.ofSeconds(secs.toLong)).toOption
+
+    case _ => None
+  } (d => s"${d.getSeconds}.${d.getNano}")
```

Comment:
Yeah, the tests should hopefully help give us some confidence about the validity (I realize they don't exist yet, but they will!)